src/Compiler/Microsoft.AspNetCore.Razor.Language/src/RazorCSharpDocument.cs

```diff
@@ -20,9 +20,11 @@ public abstract class RazorCSharpDocument
 
     public abstract RazorCodeGenerationOptions Options { get; }
 
+    public abstract RazorCodeDocument CodeDocument { get; }
+
     internal virtual IReadOnlyList<LinePragma> LinePragmas { get; }
 
-    public static RazorCSharpDocument Create(string generatedCode, RazorCodeGenerationOptions options, IEnumerable<RazorDiagnostic> diagnostics)
+    public static RazorCSharpDocument Create(RazorCodeDocument codeDocument, string generatedCode, RazorCodeGenerationOptions options, IEnumerable<RazorDiagnostic> diagnostics)
```

Comment:
I checked, and whilst Web Tools does use this type, they don't call the Create method

Reply:
I have avoided it.. see what you think